mongodb/src/mongodb/endpoint.bal

```diff
@@ -31,41 +31,45 @@ public type Client client object {
     handle datasource;
 
     # Gets called when the endpoint is being initialized during the package initialization.
-    public function __init(ClientEndpointConfig config) returns error? {
-        self.datasource = initClient(config);
+    public function __init(ClientEndpointConfig config) returns ClientError? {
+        handle | ClientError? result = initClient(config);
+        if (result is ClientError?) {
+            return result;
+        } else {
+            self.datasource = result;
+        }
```

Comment:
We can also say

```ballerina
        handle? result = check initClient(config);
        if (result is handle) {
            self.datasource = result;
        }
```

Reply:
If only if part is there, other APIs are throwing not initialised error.
```
handle? result = check initClient(config);
        if (result is handle) {
            self.datasource = result;
        } else {
            return result;
        }
```

Reply:
Fixed as mentioned in the above comment d0753b5

Reply:
Not sure if I understood the issue. In this else block, you are returning `()`, which is the same as not returning anything. So we can avoid the else block.

Are you getting an error?

Reply:
The function returning nil was a mistake. Fixed as discussed offline in 0ff6a3b